atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -42,15 +42,40 @@ private const val MIN_SUPPORTED_KGP_VERSION = "1.7.0"
 open class AtomicFUGradlePlugin : Plugin<Project> {
     override fun apply(project: Project) = project.run {
         checkCompatibility()
-        val pluginVersion = rootProject.buildscript.configurations.findByName("classpath")
-            ?.allDependencies?.find { it.name == "atomicfu-gradle-plugin" }?.version
+        val classpath = project.buildscript.configurations.getByName("classpath")
+        val pluginVersion = classpath.resolvedConfiguration.resolvedArtifacts
```

Comment:
My bad, looks like this could not work in all cases.
From [last comment in a thread](https://discuss.gradle.org/t/how-can-a-custom-gradle-plugin-determine-its-own-version/36761/4):
> However, it fails for a multi-module gradle project where the plugin is added in the root projects as apply false, then applied without a version in the subproject.

Sorry, that I've sent a link to the specific message instead of just a thread :)

I was more thinking of the approach with a `.properties` file with version in it (mentioned in [second comment](https://discuss.gradle.org/t/how-can-a-custom-gradle-plugin-determine-its-own-version/36761/3)) if we don't want to bring external plugin for such simple thing (which is fine).

Also, I've found more [recent thread](https://discuss.gradle.org/t/how-to-get-plugin-itselfs-version/46413) (Sep 2023) where there is even better suggestion without a need for a file generation, as it's the safest and conventional way.

And BTW KGP does the same!
* build setup is [here](https://github.com/JetBrains/kotlin/blob/a5ef668e3c4d15778ea3d072221c6e379b5fb4c2/libraries/tools/kotlin-gradle-plugin/build.gradle.kts#L134-L146), though most likely the task name here will be just `processResources`
* template `project.properties` file is [here](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/common/resources/project.properties)
* function to read version is [here](https://github.com/JetBrains/kotlin/blob/a5ef668e3c4d15778ea3d072221c6e379b5fb4c2/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/utils/resourceUtils.kt#L17-L24)
* example of usage is [here](https://github.com/JetBrains/kotlin/blob/a5ef668e3c4d15778ea3d072221c6e379b5fb4c2/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPluginWrapper.kt#L386-L403)
* And additionally, just may be it will also help, here is the [commit](https://github.com/JetBrains/kotlin/commit/acf8a0454fa99e402c957ce675df7338714b15f6) in KGP which adds additional property to this file, so it touches all affected places mentioned earlier (though, it's from 2019, so some code can be moved already)

Reply:
Thank you for the great navigation 🔥!) Seems, that this approach worked for me.
I've pushed a commit with `atomicfu.properties` file generation.